app/src/test/java/com/android/sample/model/network/ActivityDaoTest.kt

```diff
@@ -0,0 +1,65 @@
+package com.android.sample.model.network
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.room.Room
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import com.android.sample.model.activity.ActivityStatus
+import com.android.sample.model.activity.database.ActivityDao
+import com.android.sample.model.activity.database.AppDatabase
+import com.android.sample.resources.dummydata.activity1
+import com.android.sample.resources.dummydata.activity2
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
```

Comment:
added tests for the ActivityDao, including methods to insert and retrieve activities as well as to clear activities by status, ensuring the DAO functions correctly.